drivers/net/bnxt/bnxt_hwrm.c

```diff
@@ -301,6 +301,10 @@ int bnxt_hwrm_set_filter(struct bnxt *bp,
 	if (enables &
 	    HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_L2_OVLAN_MASK)
 		req.l2_ovlan_mask = filter->l2_ovlan_mask;
+	if (enables & HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_SRC_ID)
+		req.src_id = rte_cpu_to_le_32(filter->src_id);
+	if (enables & HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_SRC_TYPE)
+		req.src_type = rte_cpu_to_le_32(filter->src_type);
```

Comment:
Old code.. the l2_ovlan_mask and l2_ovlan are u16. but we are not doing the cpu_tole16 on the assigned values.
Do you want to fix that as well?